### PR TITLE
Enable "Dev Containers" development environment (VSCode, GitHub Codespace, etc.)

### PR DESCRIPTION
**Pull request by skkdevcraft**
## Context

Enable development via the "Dev Containers" VSCode feature. This also makes the repo compatible with the GitHub "Codespace" feature.

Working withing dev container provides better isolation than WSL2 on windows and is virtually identical on both MacOS and Windows.

## Docs change

The dev containers feature will allow to extend the [Local Setup](https://twenty.com/developers/local-setup) page with:

- Add a tab "Docker for Desktop" (Mac or Windows) in the Prerequisites
- Add a tab "Docker for Desktop" to Step 1 and Step 2 with the following instructions:

 1. Open VSCode; 
 2. Press (Ctrl + Shift + P) to open the command palette (or use the View -> Command Palette ... menu); 
 3. Then type: "Dev Containers: Clone Repository in Container Volume..."

- Add a tab "Docker for Desktop" to Step 3:

```
postgres-on-devcontainer
```

### CODE REVIEW

**Comment by skkdevcraft** [comment]
Hey everyone!

I've been switching between my Windows desktop and my M2 MacBook lately, and I wanted to share something that's really improved my workflow: using dev containers. It's been a fantastic developer experience, and I thought it might be helpful for others too!

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  app:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+
+    volumes:
+      - ../..:/workspaces:cached
+      
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
```
[the remark this answers is not in the document]
Comment (skkdevcraft) [reply .devcontainer/docker-compose.yml:13]: the container is started and stopped manually by the IDE and is not intended for production life-cycle management

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+
+    environment:
+      PGHOST: db
+      PGUSER: postgres
+      PGPASSWORD: postgres
```
[the remark this answers is not in the document]
Comment (skkdevcraft) [reply .devcontainer/docker-compose.yml:21]: [dx] the "db" container is accessible only by the "app" container as in a "private network". providing the credentials as environment variables allows us to use the ```psql``` cli directly (from the "app" container) without specifying them each time. for example ```psql --list``` should we need external access to the database server, the recommended approach is to add a container with pgadmin or similar.

**Comment by lucasbordeau** [comment]
@skkdevcraft Thank you for this proposal. 

We understand your particular need but we already chose to not use dev containers and for the moment we don't know if we want to introduce them again for everyone.

I'm closing this but we'll surely keep it in the back of our head should we decide to introduce it again !

**Pull request closed without merging**